### ISSUE TITLE
Can't select a date exactly a month away

### ISSUE THREAD

**Issue by SqAtx** [getting-things-gnome/gtg#1074]
When selecting a start date or due date from the date picker, I can't pick a date that has the same day number as today.

For example, today is 2024-03-26. If I open the date picker and go one month forward to April, the 26th is already highlighted:

![image](https://github.com/getting-things-gnome/gtg/assets/4351932/87edfe4f-f8b6-4d2a-885a-c4c8cd219ef9)

Clicking on it does nothing.

Clicking on any other day correctly populates the date.

**Comment by diegogangl** [comment]
Looking at https://docs.gtk.org/gtk4/class.Calendar.html doesn't look like there's much we can do about this. The calendar doesn't allow "no day" to be selected

**Comment by SqAtx** [comment]
OK, if you think we can't fix it within GTG, we can close the ticket. I'll look into it more and see if I can open a ticket further down the stack.

**Comment by diegogangl** [comment]
Yeah, that would be great. Esentially what we need is the day-changed signal to also be fired when clicking on a day that's already been selected. Or maybe some other signal that we can connect to

**Comment by SqAtx** [comment]
I removed this from the 0.7 milestone, because I don't think it needs to make it into that version. I think it's still a problem that needs to be fixed one way or another, so I'm keeping the issue open and will take another look later.

**Comment by pentux-GitHub** [comment]
Still reproducible on current master (0e99f945): clicking the date exactly one month away has no effect and the popover stays open, while any other date works and closes it. Likely mechanism: the GTK calendar widget pre-selects that day (it shows highlighted when the popover opens, which is #809's subject), and clicking an already-selected day emits no day-selected signal, so the handler never fires. Confirming the two issues share the same root.
